### ISSUE TITLE
Error saving credentials: error storing credentials - err: exit status 1, out: [...] connection refused`

### ISSUE THREAD

**Issue by AlliOOPsmart** [docker/docker-credential-helpers#240]
Error saving credentials: error storing credentials - err: exit status 1, out: `Post "http://ipc/registry/credstore-updated": dial unix /Users/aleonard/Library/Containers/com.docker.docker/Data/backend.sock: connect: connection refused`

What are some reasons for the connection being refused??

**Comment by thaJeztah** [comment]
Looks like this may be specific to docker desktop, not the code in this repository 🤔 . Was Docker Desktop running when you tried to authenticate? The error indicates that the docker-desktop helper may be trying to connect to the Docker Desktop API socket, and it not (yet?) running.

**Comment by AlliOOPsmart** [comment]
Solved! Thanks @thaJeztah 
I tried a few things and verified that Docker was up via the command line (`top` `ps`) but the "fix" was using the Mac search bar to open up the Docker console. On the first few tries launching -- the Docker App would open and then close immediately. Opening the Docker console kept failing until it opened successfully on the 4th try - i dont know why. After verifying the engine was running _in the Docker console_ the command to build/push my docker images worked without the `connection refused` error
